src/components/periodic-report/handlers/sync-periodic-report-to-project.handler.ts

```diff
@@ -0,0 +1,115 @@
+import { isNull, node, relation } from 'cypher-query-builder';
+import { DateTime } from 'luxon';
+import { Session } from '../../../common';
+import {
+  DatabaseService,
+  EventsHandler,
+  IEventHandler,
+  ILogger,
+  Logger,
+} from '../../../core';
+import { Project } from '../../project';
+import { ProjectUpdatedEvent } from '../../project/events';
+import { CreatePeriodicReport } from '../dto';
+import { ReportType } from '../dto/type.enum';
+import { PeriodicReportService } from '../periodic-report.service';
+
+type SubscribedEvent = ProjectUpdatedEvent;
+
+@EventsHandler(ProjectUpdatedEvent)
+export class SyncPeriodicReportToProject
+  implements IEventHandler<SubscribedEvent> {
+  constructor(
+    private readonly db: DatabaseService,
+    private readonly periodicReports: PeriodicReportService,
+    @Logger('periodicReport:sync-project') private readonly logger: ILogger
+  ) {}
+
+  async handle(event: SubscribedEvent) {
+    this.logger.debug('Project mutation, syncing periodic reports', {
+      ...event,
+      event: event.constructor.name,
+    });
+
+    const project = event.updated;
+    const previous = event.previous;
+
+    const isDateRangeChanged =
+      project.mouStart !== previous.mouStart ||
+      project.mouEnd !== previous.mouEnd;
+
+    if (isDateRangeChanged) {
+      await this.removeOldReports(project, event.session);
+      await this.syncRecords(project, event.session);
+    }
+  }
+
+  private async removeOldReports(project: Project, session: Session) {
```

Comment:
We only want to remove reports based on the date range that's being removed. This could probably be a where clause.